app/code/Magento/Catalog/Model/AbstractModel.php

```diff
@@ -235,7 +235,7 @@ public function getResourceCollection()
      * Load entity by attribute
      *
      * @param \Magento\Eav\Model\Entity\Attribute\AttributeInterface|integer|string|array $attribute
-     * @param null|string|array $value
+     * @param null|string|int|array $value
```

Comment:
Can we have `float` here as well? 
Please, take a look at `addAttributeToFilter` method of the same class. The `$condition` attribute should have the same types as hinted for `$value` in `$loadByAttribute`